### PR TITLE
Updated links from http to https in guides, docs, etc

### PR DESCRIPTION
**Pull request by abhaynikam**
Converted few links which can be converted to HTTPS after PR #35492

ref: https://github.com/rails/rails/pull/35492#issuecomment-470872285

@r? @vipulnsward

### CODE REVIEW

File: actionpack/actionpack.gemspec [hunk 1]
```diff
@@ -15,7 +15,7 @@ Gem::Specification.new do |s|
 
   s.author   = "David Heinemeier Hansson"
   s.email    = "david@loudthinking.com"
-  s.homepage = "http://rubyonrails.org"
+  s.homepage = "https://rubyonrails.org"
```
Comment (abhaynikam) [inline actionpack/actionpack.gemspec]: I have added ci skip to this, but will change in gemspec anyways affect anything?

File: actionmailer/README.rdoc [hunk 2]
```diff
@@ -128,7 +128,7 @@ Action Mailer is released under the MIT license:
 
 API documentation is at
 
-* http://api.rubyonrails.org
+* https://api.rubyonrails.org/
```
Comment (kamipo) [inline actionmailer/README.rdoc]: Why add the trailing slash to `actionmailer/README.rdoc` and `actioncable/README.md` unlike others?

File: actionpack/actionpack.gemspec (hunk 1 above)
Comment (kamipo) [reply actionpack/actionpack.gemspec]: Should not add `[ci skip]`, gemspec is definitely not doc, but Ruby code.

Comment (abhaynikam) [reply actionpack/actionpack.gemspec]: @kamipo : Removed `[ci skip]` and fixed other comments too.

File: actioncable/README.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ You can read more about Action Cable in the [Action Cable Overview](https://edge
 
 API documentation is at:
 
-* http://api.rubyonrails.org
+* https://api.rubyonrails.org/
```
Comment (kamipo) [inline actioncable/README.md]: https://github.com/rails/rails/pull/35529#discussion_r263992781

Comment (abhaynikam) [reply actioncable/README.md]: My Bad. Fixed it now.

**Comment by abhaynikam** [comment]
Thanks, @kamipo . I missed a few gemspec files to update the links to https. Is it okay if I reopen a new PR for it?

**Comment by kamipo** [comment]
Confirmed `actionview.gemspec`, `activejob.gemspec`, `activemodel.gemspec`, `activerecord.gemspec`, `activestorage.gemspec`, `activesupport.gemspec`, and `railties/RDOC_MAIN.rdoc`.
Please do that.

**Pull request merged**